lib/PoolCluster.js

```diff
@@ -124,17 +128,35 @@ PoolCluster.prototype._getNode = function(id) {
 };
 
 PoolCluster.prototype._increaseErrorCount = function(node) {
+  var self = this;
+
   if (++node.errorCount >= this._removeNodeErrorCount) {
     var index = this._serviceableNodeIds.indexOf(node.id);
     if (index !== -1) {
       this._serviceableNodeIds.splice(index, 1);
-      delete this._nodes[node.id];
-
       this._clearFindCaches();
 
-      node.pool.end();
+      if ((this._restoreNodeTimeout > 0) && !node.restoreTimeoutID) {
+        node.online = false;
+        this.emit('offline', node.id);
+
+        node.restoreTimeoutID = setTimeout(function() {
+          delete node.restoreTimeoutID;
```

Comment:
Set back to 0 or something rather than deleting and changing the hidden class of the node.